### PR TITLE
documentation field added to ContractDefinition-Node

### PR DESCRIPTION
**Pull request by djudjuu**
adresses #2243

### CODE REVIEW

**Comment by axic** [comment]
Can you add a test case to `ASTJSON` which has a non-null documentation node?

File: test/libsolidity/JSONCompiler.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ BOOST_AUTO_TEST_CASE(basic_compilation)
 	BOOST_CHECK(dev::jsonCompactPrint(result["sources"]["fileA"]["AST"]) ==
                 "{\"attributes\":{\"absolutePath\":\"fileA\",\"exportedSymbols\":{\"A\":[1]}},"
                 "\"children\":[{\"attributes\":{\"baseContracts\":[null],\"contractDependencies\":[null],"
-                "\"contractKind\":\"contract\",\"fullyImplemented\":true,\"linearizedBaseContracts\":[1],"
+		"\"contractKind\":\"contract\",\"documentation\":null,\"fullyImplemented\":true,\"linearizedBaseContracts\":[1],"
```
Comment (pirapira) [inline test/libsolidity/JSONCompiler.cpp]: The new indentation seems to follow the coding standard, but the line stands out.  Maybe in a separate commit, you can fix the indentation around here.

**Comment by chriseth** [comment]
@djudjuu please add a test that has a non-null documentation node, and also one that has a line break in the documentation.

**Comment by djudjuu** [comment]
@chriseth  there already is one non-null documentation-test in ASTJSON.cpp. do you want me to add another one?

**Comment by axic** [comment]
@djudjuu would be nice to have a dedicated test in `test/libsolidity/ASTJSON.cpp`

Also can you add tests for non-legacy mode?

**Comment by axic** [comment]
@djudjuu the dedicated was in reflection to @chriseth's comment. A test in `test/ASTJSON` for empty documentation.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
@chriseth @djudjuu I think we should still add that test 😉
